src/main/java/org/jabref/logic/cleanup/ConvertToBiblatexCleanup.java

```diff
@@ -39,8 +41,19 @@
                 entry.clearField(StandardField.YEAR).ifPresent(changes::add);
                 entry.clearField(StandardField.MONTH).ifPresent(changes::add);
             });
+        } else {
+            // If the year from date field is filled and equal to year it should be removed the year field
+            entry.getFieldOrAlias(StandardField.DATE).ifPresent(date -> {
+                Optional<Date> newDate = Date.parse(date);
+                Optional<Date> checkDate = Date.parse(entry.getFieldOrAlias(StandardField.YEAR),
+                        entry.getFieldOrAlias(StandardField.MONTH), Optional.empty());
+
```

Comment:
Just a quick note,
The Optional equals also returns true if both are empty.
That could be the case when there is a value in all fields, but it's invalid., eg maybe some random letters.
Could you please add a test case for that as well. Then you should have all cases covered. I think in practice this doesn't matter but code wise it would be nice to have this edge case covered as well. 

Otherwise it's looking good to me! Thanks again for your ongoing contribution 

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#equals-java.lang.Object-

Reply:
@Siedlerchr You are right, if both fields are empty returns true. I added a new test case to cover it.